runtime/src/rpc.rs

```diff
@@ -1580,9 +1449,11 @@ impl VaultRegistryPallet for InterBtcParachain {
 
     /// Fetch all active vaults.
     async fn get_all_vaults(&self) -> Result<Vec<InterBtcVault>, Error> {
-        let mut vaults = Vec::new();
         let head = self.get_latest_block_hash().await?;
-        let mut iter = self.api.storage().vault_registry().vaults_iter(head).await?;
+        let key_addr = metadata::storage().vault_registry().vaults_root();
+        let mut iter = self.api.storage().iter(key_addr, 10, head).await?;
```

Comment:
What is this `10`? You use it in multiple places -> let's use a constant with a descriptive name